utilities/transactions/write_unprepared_txn_db.cc

```diff
@@ -367,29 +365,49 @@ Iterator* WriteUnpreparedTxnDB::NewIterator(const ReadOptions& options,
   // TODO(lth): Refactor so that this logic is shared with WritePrepared.
   constexpr bool ALLOW_BLOB = true;
   constexpr bool ALLOW_REFRESH = true;
-  std::shared_ptr<ManagedSnapshot> own_snapshot = nullptr;
```

Comment:
If we end up reverting the change of getting GetSnapshot, are we going to put back the own_snapshot here?

Reply:
Yes, own_snapshot will have to stay if we can't rely on the txn always holding a snapshot.